### PR TITLE
Add event detector

### PR DESCRIPTION
**Pull request by FedericoCeratto**
Add event detector and RSS feed generation
Implements #229
Related to #213 and https://github.com/ooni/spec/issues/144

### CODE REVIEW

File: af/fastpath/debian/detector.service [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+# Hardening
+CapabilityBoundingSet=CAP_SETUID CAP_SETGID
+SystemCallFilter=~@clock @debug @cpu-emulation @keyring @module @mount @obsolete @raw-io @reboot @swap
+NoNewPrivileges=yes
+PrivateDevices=yes
+PrivateTmp=yes
+ProtectHome=yes
+#ProtectHome=read-only
+ProtectSystem=strict
+ProtectSystem=full
```
Comment (bassosimone) [inline af/fastpath/debian/detector.service]: Is `ProtectSystem=full` overriding or augmenting `ProtectSystem=strict`?

File: af/fastpath/fastpath/detector.py [hunk 2]
```diff
@@ -0,0 +1,520 @@
...
+    FROM measurement
+    JOIN report ON report.report_no = measurement.report_no
+    JOIN input ON input.input_no = measurement.input_no
+    WHERE measurement_start_time >= %(start_date)s
+    AND probe_cc = %(cc)s
+    AND test_name = %(test_name)s
+    AND input = %(inp)s
+
+    ORDER BY measurement_start_time
+    """
```
Comment (bassosimone) [inline af/fastpath/fastpath/detector.py]: I assume your end goal is to use this function to implement also the one above?

File: af/fastpath/debian/control [hunk 3]
```diff
@@ -14,6 +14,7 @@ Depends: ${misc:Depends},
  ${python3:Depends},
  python3-paramiko,
  python3-statsd,
- python3-systemd
+ python3-systemd,
+ nginx
```
Comment (bassosimone) [inline af/fastpath/debian/control:31]: Is this redundant, considering that we are using Ansible to deploy hosts?

**Review by bassosimone — COMMENTED** [review]
Hello @FedericoCeratto, here are my preliminary comments!

I like the principle of using Debian functionality as much as possible! I am unfortunately not familiar enough yet with our Ansible infrastructure, so I've asked a question regarding that.

I believe it would be beneficial to factor the bits where we perform measurement analysis in a different module with unit tests, so it would be easier (at least for me) to understand and visualize what is the typical input and the typical output. (That is, I'm not advocating for full coverage, even though this is nice, but rather to have unit tests as live examples of building blocks functionality.)

Thank you!

File: af/fastpath/debian/detector.service (hunk 1 above)
Comment (FedericoCeratto) [reply af/fastpath/debian/detector.service]: It's overriding it, I should remove the other.
https://www.freedesktop.org/software/systemd/man/systemd.exec.html#Sandboxing

**Comment by hellais** [comment]
I am a bit concerned of the approach being so much reliant on debian packages for deploying stuff.

I fear this may lead to bits of our infrastructure using a certain pattern and others using a different one, making it necessary to know more bits of the system in order to do a new deployment.

I guess that if you publish the debian packages to some location and then the ansible playbook fetches a prebuilt packages, that is ok, though I would like to understand better what is the reason to go away from the current ansible based infrastructure.

I think it's easier to reason about a system (and make cross-cutting changes) when the patterns used in all the infrastructure are few and consistent.

I don't think it's necessary to use Docker (though it does make our life significantly easier in many regards), but I would like to better understand what direction you are thinking of going by introducing debian packaging in here.

File: af/fastpath/fastpath/detector.py (hunk 2 above)
Comment (FedericoCeratto) [reply af/fastpath/fastpath/detector.py]: Those are 2 different use cases: filtering by test_name/cc/input or getting everything.

File: af/fastpath/debian/control (hunk 3 above)
Comment (FedericoCeratto) [reply af/fastpath/debian/control:31]: The dependencies are also used to run devel version on workstations,  run unit/integ tests and ensure that we are really deploying the required deps at runtime

**Review by hellais — APPROVED** [review]
We further discussed the deb based deployed and I think we have found a solution which can work.

For me this branch is good to be landed in master.

**Pull request merged**